### PR TITLE
bump kube-addon-mgr base

### PR DESCRIPTION
**Pull request by sozercan**
Signed-off-by: Sertac Ozercan <sozercan@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Existing base image (debian-base:v1.0.1) is debian `stretch` which is EOL and does not build anymore

```shell
$ make build
...
 => ERROR [2/5] RUN clean-install bash                                                                                                                                             3.2s
------                                                                                                                                                                                  
 > [2/5] RUN clean-install bash:      
...
2.713 Err:10 http://deb.debian.org/debian stretch/main amd64 Packages
2.713   404  Not Found
2.885 Err:11 http://deb.debian.org/debian stretch-updates/main amd64 Packages
2.885   404  Not Found
2.911 Ign:12 http://deb.debian.org/debian stretch-updates/main all Packages
2.914 Reading package lists...
2.918 W: The repository 'http://security.debian.org/debian-security stretch/updates Release' does not have a Release file.
2.918 W: The repository 'http://deb.debian.org/debian stretch Release' does not have a Release file.
2.918 W: The repository 'http://deb.debian.org/debian stretch-updates Release' does not have a Release file.
2.918 E: Failed to fetch http://security.debian.org/debian-security/dists/stretch/updates/main/binary-amd64/Packages  404  Not Found
2.918 E: Failed to fetch http://deb.debian.org/debian/dists/stretch/main/binary-amd64/Packages  404  Not Found
2.918 E: Failed to fetch http://deb.debian.org/debian/dists/stretch-updates/main/binary-amd64/Packages  404  Not Found
2.918 E: Some index files failed to download. They have been ignored, or old ones used instead.
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/addons/addon-manager/Makefile [hunk 1]
```diff
@@ -15,10 +15,10 @@
 IMAGE=gcr.io/k8s-staging-addon-manager/kube-addon-manager
 ARCH?=amd64
 TEMP_DIR:=$(shell mktemp -d)
-VERSION=v9.1.7
-KUBECTL_VERSION?=v1.27.1
+VERSION=v9.1.8
+KUBECTL_VERSION?=v1.28.2
 
-BASEIMAGE=registry.k8s.io/debian-base-$(ARCH):v1.0.1
+BASEIMAGE=registry.k8s.io/debian-base-$(ARCH):v2.0.0
```
Comment (pacoxu) [inline cluster/addons/addon-manager/Makefile:21]: https://github.com/kubernetes/kubernetes/blob/09200e9c92784f6fb9210e6287e306649a731673/cluster/images/etcd/Makefile#L95-L96 is using the `bookworm-v1.0.0`. 

https://github.com/kubernetes/kubernetes/blob/09200e9c92784f6fb9210e6287e306649a731673/build/dependencies.yaml#L145-L177
This needs an update if we want to migrate to v2.0.0.

Comment (sozercan) [reply cluster/addons/addon-manager/Makefile]: @pacoxu thanks! i updated to use `debian-base:bookworm-v1.0.0` if that works

**Comment by djsly** [comment]
@sozercan / @pacoxu anything blocking here ?

File: cluster/addons/addon-manager/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+### Version 9.1.8 (Tue September 19 2023 Sertac Ozercan <sozercan@gmail.com>)
+- Update kubectl to v1.28.2.
+- Update base image to debian-base:v2.0.0.
```
Comment (pacoxu) [inline cluster/addons/addon-manager/CHANGELOG.md:3]:
```suggestion
- Update base image to debian-base:bookworm-v1.0.0.
```

Comment (sozercan) [reply cluster/addons/addon-manager/CHANGELOG.md]: updated, thank you!

**Comment by AshutoshNirkhe** [comment]
Anything still blocking this PR ?

**Pull request closed without merging**